lib/console/gateways/gateways.ex

```diff
@@ -53,10 +54,36 @@ defmodule Console.Gateways do
       {:error, %Ecto.Changeset{}}
 
   """
-  def create_gateway(attrs \\ %{}) do
-    %Gateway{}
-    |> Gateway.changeset(attrs)
-    |> Repo.insert()
+  def create_gateway(attrs \\ %{}, attempt \\ 1) do
+    token = :crypto.strong_rand_bytes(4)
```

Comment:
this token generation (and also maybe the retry logic?) probably belongs in the hardware identifier schema file